### ISSUE TITLE
Error when using pipe for union types

### ISSUE THREAD

**Issue by LysanderGG** [mkorpela/overrides#115]
The following example using `Union[int, str]` works fine.

```python
from typing import Union

from overrides import override


class A:
    def f(self) -> Union[int, str]:
        return 1


class B(A):
    @override
    def f(self) -> int:
        return 1
```

but the following example replacing `Union[int, str]` with `int | str` fails.

```python
from overrides import override


class A:
    def f(self) -> int | str:
        return 1


class B(A):
    @override
    def f(self) -> int:
        return 1
```

The error is as follows:

```
Traceback (most recent call last):
  File ".../.pyenv/versions/3.10.10/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File ".../.pyenv/versions/3.10.10/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File ".../example.py", line 9, in <module>
    class B(A):
  File ".../example.py", line 11, in B
    def f(self) -> int:
  File ".../venv/lib/python3.10/site-packages/overrides/overrides.py", line 143, in override
    return _overrides(method, check_signature, check_at_runtime)
  File ".../venv/lib/python3.10/site-packages/overrides/overrides.py", line 170, in _overrides
    _validate_method(method, super_class, check_signature)
  File ".../venv/lib/python3.10/site-packages/overrides/overrides.py", line 189, in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
  File ".../venv/lib/python3.10/site-packages/overrides/signature.py", line 102, in ensure_signature_is_compatible
    ensure_return_type_compatibility(super_type_hints, sub_type_hints, method_name)
  File ".../venv/lib/python3.10/site-packages/overrides/signature.py", line 302, in ensure_return_type_compatibility
    raise TypeError(
TypeError: B.f: return type `<class 'int'>` is not a `int | str`.
```

Is it a known limitation or an issue?

**Comment by mkorpela** [comment]
Seems like an issue. Thanks for the report!

**Comment by mkorpela** [comment]
Seems the problem is that typing.Union and types.UnionType are not the same thing.

**Comment by mkorpela** [comment]
Fixed in 7.4.0
